src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp

```diff
@@ -370,28 +370,33 @@ namespace OpenMS
   {
     
     double ratio = 0.0;
-    if (component_1.metaValueExists(feature_name) && component_2.metaValueExists(feature_name))
-    {
-      double feature_1 = component_1.getMetaValue(feature_name);
-      double feature_2 = component_2.getMetaValue(feature_name);
-      ratio = feature_1/feature_2;
-      
-    }
-    else if (feature_name == "intensity")
+    if (feature_name == "intensity" && component_1.metaValueExists("native_id") && component_2.metaValueExists("native_id"))
     {
       const double feature_1 = component_1.getIntensity();
       const double feature_2 = component_2.getIntensity();
       ratio = feature_1 / feature_2;
     }
+    else if (feature_name == "intensity" && component_1.metaValueExists("native_id"))
+    {
+      LOG_DEBUG << "Warning: no IS found for component " << component_1.getMetaValue("native_id") << ".";
```

Comment:
in this if branch, the omponent_2.metaValueExists("native_id") is false. Though the debug message talks about component 1. Is this correct?

Reply:
Ah I get it.